x/btcstaking/types/metrics.go

```diff
@@ -0,0 +1,103 @@
+package types
+
+import (
+	"github.com/cosmos/cosmos-sdk/telemetry"
+	"github.com/hashicorp/go-metrics"
+)
+
+// performance oriented metrics measuring the execution time of each message
+const (
+	MetricsKeyCreateFinalityProvider    = "create_finality_provider"
+	MetricsKeyCreateBTCDelegation       = "create_btc_delegation"
+	MetricsKeyAddCovenantSigs           = "add_covenant_sigs"
+	MetricsKeyBTCUndelegate             = "btc_undelegate"
+	MetricsKeySelectiveSlashingEvidence = "selective_slashing_evidence"
+)
+
+// Metrics for monitoring finality providers and BTC delegations
+const (
+	// MetricsKeyFinalityProviders is the key of the gauge recording the number
+	// of {active, inactive} finality providers, and the key of the counter
+	// recording the number of slashed finality providers
+	MetricsKeyFinalityProviders = "finality_providers"
+	// MetricsKeyBTCDelegations is the key of the gauge recording the number of
+	// {pending, active, unbonded} BTC delegations, and the key of the counter
+	// recording the number of slashed BTC delegations
+	MetricsKeyBTCDelegations = "btc_delegations"
+	// MetricsKeyStakedBitcoins is the key of the gauge recording the total
+	// amount of Bitcoins staked under active finality providers
+	MetricsKeyStakedBitcoins = "staked_bitcoins"
+)
+
+// RecordActiveFinalityProviders records the number of active finality providers.
+// It is triggered upon recording voting power table.
+func RecordActiveFinalityProviders(num int) {
```

Comment:
```suggestion
func RecordActiveFinalityProviders(num uint64) {
```

In general, should we change the other `int`s in this file to `uint64`s as well?

Reply:
Given that this is the number of finality providers, probably `int` is more suitable. For example `len()` returns an int as well. Wdyt?

Reply:
Hm no strong opinions here. My only concern is that we already know that these are unsigned integers so we could more strongly type this, but np either way.